### PR TITLE
Added integration tests for error scenarios and added error observability to plugin

### PR DESCRIPTION
**Pull request by nathandotleeathpe**
I added integration tests simulating driver errors at each DWS stage. The feature file contains a comment detailing how the tests need to change if `Flags=TeardownFailure` is configured.

Driver errors are now returned to Slurm, allowing the errors to show up in the `SystemComment` field in `scontrol` output. This was written using a json path to return all DriverStatus entries for the current watch state and then filtering through those entries to find Errors. The json path was designed to handle multiline driver errors. Also note that lua iterators will return "nil" if you have reached the end of their queue.

The unit tests were updated to cover the driver error handling. Previously, the unit tests were written to also run against a live k8s and slurm environment. The integration tests remove the need for this feature, so I removed it. I also removed the old CRD validator tool for the same reason.

### CODE REVIEW

File: testsuite/unit/src/burst_buffer/dws-test.lua [hunk 1]
```diff
@@ -1066,23 +925,62 @@ describe("Slurm API", function()
...
 		call_bb_setup()
 		call_bb_state("DataIn")
+		call_bb_get_status("DataIn", "Completed")
 		call_bb_state("PreRun")
 		call_bb_state("PostRun")
 		call_bb_state("DataOut")
 		call_bb_teardown()
 	end)
 
+	insulate("reports driver error(s)", function()
```
Comment (roehrich-hpe) [inline testsuite/unit/src/burst_buffer/dws-test.lua:935]: When I experimented with insulate() a few months ago, I found that it has a suprisingly narrow purpose.  It probably does nothing special for us here.  If you switch it to context() does it break this code block or affect the next context()?

Comment (nathandotleeathpe) [reply testsuite/unit/src/burst_buffer/dws-test.lua:935]: Yeah, I was hoping `insulate` would do more. `context` works just fine in this case. I'll replace it.

**Comment by roehrich-hpe** [comment]
Would you please update the dws-test-driver and slurm-docker-cluster submodules with this PR?  When they're not pointing into 'main' then the github cli tool has trouble checking out this PR:
```
$ gh pr checkout 30 
$ git submodule update --init --recursive
[...]
fatal: Fetched in submodule path 'testsuite/submodules/slurm-docker-cluster', but it did not contain 2f85cf28023416c0382a5a0e31a885b24f67a2bd. Direct fetching of that commit failed.
```
Maybe that's a commit in one of your personal forks?

File: testsuite/unit/src/burst_buffer/dws-test.lua (hunk 1 above)
Comment (roehrich-hpe) [reply testsuite/unit/src/burst_buffer/dws-test.lua]: Here's one place where it does something useful:
https://github.com/DataWorkflowServices/dws-slurm-bb-plugin/blob/main/testsuite/unit/src/burst_buffer/dws-test.lua#L1103-L1112

**Comment by roehrich-hpe** [comment]
>

**Comment by roehrich-hpe** [comment]
When you update the submodules, you should leave 'dws' at v0.0.6, until we snap another release of it.

File: testsuite/unit/src/burst_buffer/dws-test.lua [hunk 2]
```diff
@@ -442,10 +393,27 @@ describe("The dws library", function()
...
+			io.popen:clear()
+
+			assert.is_true(done, err)
+			assert.is_equal(status["desiredState"], state)
+			assert.is_equal(status["currentState"], state)
+			assert.is_equal(status["status"], "Completed")
+		end
+
+		-- Handle errors returned from DWS during state transition
+		local wait_for_state_errors = function(state)
```
Comment (roehrich-hpe) [inline testsuite/unit/src/burst_buffer/dws-test.lua:407]: This function is not used, and it's identical to the one above it.

File: testsuite/integration/src/features/test_dws_states.feature [hunk 3]
```diff
@@ -36,82 +36,73 @@ Feature: Data Workflow Services State Progression
...
-            | DataOut  | false      |
-            | Proposal | true       |
-            | Setup    | true       |
-            | DataIn   | true       |
-            | PreRun   | true       |
-            | PostRun  | true       |
-            | DataOut  | true       |
\ No newline at end of file
+        Then the job's system comment contains the following:
+            TEST ERROR
```
Comment (roehrich-hpe) [inline testsuite/integration/src/features/test_dws_states.feature:107]: TESTunderscoreERROR

**Review by roehrich-hpe — APPROVED** [review]

**Pull request merged**